starky/src/util.rs

```diff
@@ -0,0 +1,16 @@
+use itertools::Itertools;
+use plonky2::field::field_types::Field;
+use plonky2::field::polynomial::PolynomialValues;
+use plonky2::util::transpose;
+
+/// A helper function to transpose a row-wise trace and put it in the format that `prove` expects.
+pub fn trace_rows_to_poly_values<F: Field, const COLUMNS: usize>(
+    trace_rows: Vec<[F; COLUMNS]>,
+) -> Vec<PolynomialValues<F>> {
+    let trace_row_vecs = trace_rows.iter().map(|row| row.to_vec()).collect_vec();
+    let trace_col_vecs: Vec<Vec<F>> = transpose(&trace_row_vecs);
+    trace_col_vecs
+        .into_iter()
+        .map(|column| PolynomialValues::new(column))
+        .collect()
```

Comment:
nit: I think the corresponding method in the Plonky2 prover uses `par_iter`. It probably doesn't change much in terms of performance but maybe it's best if they match.

Reply:
Hm it seems the Plonky2 code has to clone columns because it uses the witness grid for something else later, and I think copying memory can be faster with multiple threads.

In this case I think the `Vec` should be taking ownership of the array rather than copying, making this almost free, but I'm not certain. I'll add a TODO to check.

Reply:
I did a bit of timing, it seems to take around 0.04s either way (serial or parallel iterator) in our `SystemZero` test, though it was very noisy so hard to compare.

Reply:
Slight correction, it's converting arrays to Vecs that's taking 0.04s - I guess it does need to copy because of the way arrays are stored as contiguous values (might be avoidable if we box them?).

Converting to `PolynomialValues` seems to be free - makes sense since it's a trivial wrapper around the `Vec`, so shouldn't change the way data is stored.

Reply:
Oh ok makes sense. In that case I agree it's fine not to use rayon.